python/pyarrow/tests/test_table.py

```diff
@@ -3622,7 +3638,7 @@ def test_recordbatch_non_cpu(cuda_context, cpu_recordbatch, cuda_recordbatch,
         cuda_recordbatch.cast(new_schema)
 
     # drop_null() test
-    null_col = pa.array([-2, -1, 0, 1, 2],
+    null_col = pa.array([1] * cuda_recordbatch.num_rows,
                         mask=[True, False, True, False, True]).copy_to(
```

Comment:
Here it is a bit strange to generalize this with `num_rows`, because the `mask` is still (and has to be, I think, to have both true and falses) hardcoded